src/fraction/unicode_str_io.rs

```diff
@@ -0,0 +1,592 @@
+                            '6' => '₆',
+                            '7' => '₇',
+                            '8' => '₈',
+                            '9' => '₉',
+                            '0' => '₀',
+                            c => c,
+                        })
+                        .collect::<String>(),
+                )
+            }
+            _ => write!(f, "{}", self.0),
+        }
+    }
+}
+
+impl<T: Clone + Integer + fmt::Display> GenericFraction<T> {
+    /// Display the fraction using FRACTION SLASH '\u{2044}' '⁄'
+    /// If you have font support, this is the way Unicode wants to display fractions
+    /// ```
+    /// use fraction::Fraction;
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(1u8,2u8).get_unicode_display()),
+    ///   "1⁄2"
+    /// );
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(3u8,2u8).get_unicode_display()),
+    ///   "3⁄2"
+    /// );
+    /// ```
+    pub fn get_unicode_display(&self) -> UnicodeDisplay<'_, T> {
+        UnicodeDisplay(self)
+    }
+}
+
+impl<T: Clone + Integer + fmt::Display + From<u8>> GenericFraction<T> {
+    /// Parse a unicode string
+    /// The string can be:
+    /// - A normal fraction e.g. "1/2"
+    /// - A vulgar fraction e.g. '½'
+    /// - ~A mixed vulgar fraction "1½"~
+    /// - A unicode fraction e.g. "1⁄2" where '⁄' can be any of:
+    ///   - '/' ASCII SOLIDUS
+    ///   - '⁄' FRACTION SLASH
+    ///   - '∕' DIVISION SLASH
+    ///   - '÷' DIVISION SIGN
+    /// - A mixed unicode fraction: "1\u{2063}1⁄2": 1⁤1⁄2
+    ///   - '\u{2064}' INVISIBLE PLUS
+    ///   - '\u{2063}' INVISIBLE SEPARATOR
+    ///   - NOT ~'\u{2062}' INVISIBLE TIMES~
+    /// - A super-subscript fraction "¹/₂"
+    /// - A mixed super-subscript fraction  "1¹/₂"
+    ///
+    /// Focus is on being lenient towards input rather than being fast.
+    /// ```
+    /// use fraction::Fraction;
+    /// let v = vec![
+    ///  ("1/2", Fraction::new(1u8,2u8)),
+    ///  ("-1/2", Fraction::new_neg(1u8,2u8)),
+    ///  ("½", Fraction::new(1u8,2u8)),
+    ///  // ("1½", Fraction::new(1u8,2u8)),
+    ///  // ("-1½", Fraction::new_neg(1u8,2u8)),
+    ///  ("1⁄2", Fraction::new(1u8,2u8)),
+    ///  ("-1⁄2", Fraction::new_neg(1u8,2u8)),
+    ///  ("1⁤1⁄2", Fraction::new(3u8,2u8)),
+    ///  ("-1⁤1⁄2", Fraction::new_neg(3u8,2u8)),
+    ///  ("¹/₂", Fraction::new(1u8,2u8)),
+    ///  ("-¹/₂", Fraction::new_neg(1u8,2u8)),
+    ///  ("1¹/₂", Fraction::new(3u8,2u8)),
+    ///  ("-1¹/₂", Fraction::new_neg(3u8,2u8)),
+    /// ];
+    /// for (f_str, f) in v {
+    ///   assert_eq!(Fraction::from_unicode_str(f_str), Ok(f))
+    /// }
+    /// ```
+    pub fn from_unicode_str(input: &str) -> Result<Self, ParseError> {
+        let s: &str;
+        let sign = if input.starts_with('-') {
+            s = &input.strip_prefix('-').unwrap();
+            Sign::Minus
+        } else if input.starts_with('+') {
+            s = &input.strip_prefix('+').unwrap();
+            Sign::Plus
```

Comment:
these double checks are here, because Clippy complained about manual stripping. Hopefully, the compiler gets rid of it